src/portalnet/protocol.rs

```diff
@@ -32,11 +33,75 @@ impl Default for PortalnetConfig {
 pub const PROTOCOL: &str = "portal";
 
 pub struct PortalnetProtocol {
-    discovery: Discovery,
+    discovery: Arc<Discovery>,
     data_radius: U256,
+}
+
+pub struct PortalnetEvents {
+    data_radius: U256,
```

Comment:
I still think `data_radius` can be removed from `PortalnetProtocol` (hm, looking back, the comment might be lost because it was made on a now-deleted commit). Fine to put that cleanup in the next PR...